flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/rest/serde/ResultInfo.java

```diff
@@ -64,12 +67,21 @@ public class ResultInfo {
         this.rowFormat = rowFormat;
     }
 
-    public static ResultInfo createResultInfo(ResultSet resultSet, RowFormat rowFormat) {
+    public static ResultInfo createResultInfo(
+            ResultSet resultSet,
+            RowFormat rowFormat,
+            @Nullable RowDataLocalTimeZoneConverter timeZoneConverter) {
         Preconditions.checkArgument(resultSet.getResultType() != ResultSet.ResultType.NOT_READY);
         List<RowData> data = resultSet.getData();
 
         switch (rowFormat) {
             case JSON:
+                if (timeZoneConverter != null && timeZoneConverter.hasTimeZoneData()) {
```

Comment:
Do we need to consider other row formats such as `PLAIN_TEXT`?

Reply:
Currently there are only tow formats: `PAIN_TEXT` and `JSON`. There will be zone id in `RowDataToStringConverterImpl` and the converter will use `CastExecutor` with the zone id to convert time to string, so it's no need to do it for format `PAIN_TEXT`.